File: Tutorials/VB/Tutorials/Tutorials/Tutorial09.vb
Imports Excel = NetOffice.ExcelApi

Public Class Tutorial09
    Implements ITutorial

    Dim _hostApplication As IHost

#Region "ITutorial Member"

    Public Sub Run() Implements TutorialsBase.ITutorial.Run

        ' In some situations you want use NetOffice with a already running application.
        ' this examples show you how its possible.

        ' GetActiveInstance take the first instance in memory
        Dim excelApplication As Excel.Application = Excel.Application.GetActiveInstance()

        ' another method is GetActiveInstances:
        ' 
        ' GetActiveInstances takes all instances in memory. dont forget to dispose the instances.
        '            
        ' Dim excelApplications() As Excel.Application = Excel.Application.GetActiveInstance()

        excelApplication.Quit()
        excelApplication.Dispose()

        _hostApplication.ShowFinishDialog()

    End Sub

    Public ReadOnly Property Caption As String Implements TutorialsBase.ITutorial.Caption
        Get
            Return "Tutorial09"
        End Get
    End Property

    Public ReadOnly Property Description As String Implements TutorialsBase.ITutorial.Description
        Get
            Return IIf(_hostApplication.LCID = 1033, "Create a NetOffice Excel Application Object with given COM Proxy", "Eine NetOffice Excel Application Objekt basierend auf einem COM Proxy erstellen")
        End Get
    End Property

    Public Sub Connect(ByVal hostApplication As TutorialsBase.IHost) Implements TutorialsBase.ITutorial.Connect

        _hostApplication = hostApplication

    End Sub

    Public Sub ChangeLanguage(ByVal lcid As Integer) Implements TutorialsBase.ITutorial.ChangeLanguage

    End Sub

    Public Sub Disconnect() Implements TutorialsBase.ITutorial.Disconnect

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements TutorialsBase.ITutorial.Panel
        Get
            Return Nothing
        End Get
    End Property


    Public ReadOnly Property Uri As String Implements TutorialsBase.ITutorial.Uri
        Get
            Return IIf(_hostApplication.LCID = 1033, "http://netoffice.codeplex.com/wikipage?title=Tutorial09_EN_VB", "http://netoffice.codeplex.com/wikipage?title=Tutorial09_DE_VB")
        End Get
    End Property

#End Region

End Class

File: Tutorials/C#/Tutorials/Tutorials/Tutorial09.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using TutorialsBase;

using NetOffice;
using Excel = NetOffice.ExcelApi;

namespace TutorialsCS4
{
    public partial class Tutorial09 : ITutorial
    {
        IHost _hostApplication;

        #region ITutorial Member

        public void Run()
        {
            // In some situations you want use NetOffice with an existing proxy, its typical for COM Addins.
            // this examples show you how its possible

            // we create a native Excel proxy
            Type excelType = Type.GetTypeFromProgID("Excel.Application");
            object excelProxy = Activator.CreateInstance(excelType);
            
            // we create an Excel Application object with the proxy as parameter,
            // excel is now under control by NetOffice
            Excel.Application excelApplication = new Excel.Application(null, excelProxy);

            excelApplication.Quit();
            excelApplication.Dispose();

            _hostApplication.ShowFinishDialog();
        }

        public void Connect(IHost hostApplication)
        {
            _hostApplication = hostApplication;
        }

        public void Disconnect()
        {

        }

        public void ChangeLanguage(int lcid)
        {

        }

        public string Uri
        {
            get { return _hostApplication.LCID == 1033 ? "http://netoffice.codeplex.com/wikipage?title=Tutorial09_EN_CS" : "http://netoffice.codeplex.com/wikipage?title=Tutorial09_DE_CS"; }
        }

        public string Caption
        {
            get { return "Tutorial09"; }
        }


        public string Description
        {
            get { return _hostApplication.LCID == 1033 ? "Create a NetOffice Application with given COM Proxy" : "Eine NetOffice Application basierend auf einem COM Proxy erstellen"; }
        }

        public UserControl Panel
        {
            get { return null; }
        }

        #endregion
    }
}
